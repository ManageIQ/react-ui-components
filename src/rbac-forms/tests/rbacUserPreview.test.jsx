import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RbacUserPreview from '../rbacUserPreview';

describe('RbacUserPreview component', () => {
  const user = {
    name: 'Administrator',
    userid: 'Admin',
    email: 'mail@mail.com',
    current_group: {
      label: 'EvmGroup-super_administrator',
      onClick: () => console.log('Current group clicked'),
    },
    groups: [{
      label: 'Cloud-Operators',
      icon: 'group',
      groupId: '1',
      onClick: () => console.log('Group click'),
    }, {
      label: 'Cloud-Users',
      icon: 'group',
      groupId: '2',
      onClick: () => console.log('Group click'),
    }],
    role: {
      label: 'Cloud-Users',
      onClick: () => console.log('Current role click'),
    },
  };

  it('Should render correctly', () => {
    const tree = shallow(<RbacUserPreview user={user} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
