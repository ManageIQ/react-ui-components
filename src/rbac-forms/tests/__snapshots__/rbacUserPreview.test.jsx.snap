// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RbacUserPreview component Should render correctly 1`] = `
<div
  className="form-horizontal rbac-user-preview"
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <PreviewRow
      label="Full Name"
    >
      <p
        className="form-control-static"
      >
        Administrator
      </p>
    </PreviewRow>
    <PreviewRow
      label="Username"
    >
      <p
        className="form-control-static"
      >
        Admin
      </p>
    </PreviewRow>
    <PreviewRow
      label="E-mail Address"
    >
      <p
        className="form-control-static"
      >
        mail@mail.com
      </p>
    </PreviewRow>
    <PreviewRow
      label="Current group"
    >
      <p
        className="form-control-static"
      >
        <Icon
          name="group"
          type="fa"
        />
        <a
          className="pointer"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          EvmGroup-super_administrator
        </a>
      </p>
    </PreviewRow>
    <PreviewRow
      label="Groups"
    >
      <p
        className="form-control-static"
        key="1"
      >
        <Icon
          name="group"
          type="fa"
        />
        <a
          className="pointer"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          Cloud-Operators
        </a>
      </p>
      <p
        className="form-control-static"
        key="2"
      >
        <Icon
          name="group"
          type="fa"
        />
        <a
          className="pointer"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          Cloud-Users
        </a>
      </p>
    </PreviewRow>
    <PreviewRow
      label="Role"
    >
      <p
        className="form-control-static"
      >
        <Icon
          name="user"
          type="fa"
        />
        <a
          className="pointer"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          Cloud-Users
        </a>
      </p>
    </PreviewRow>
  </Grid>
</div>
`;
